### PR TITLE
fix: make the next button in onboarding fixed (#3656)

### LINKED ISSUE #3656: Onboarding: Next button disappears when the interface is too small.
### Is your enhancement related to a problem? Please describe

When the interface / PD is too small, the next buttons disappear and I'm unable to find them unless I increase the size of Podman Desktop. 

https://github.com/containers/podman-desktop/assets/6422176/50d8e867-e05e-4298-bd4f-6592e54e6bbc

### Describe the solution you'd like

Being able to see the buttons regardless how small PD is.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by lstocchi**
### What does this PR do?

It makes the bar with the next/cancel buttons fixed to the bottom and 80% transparency.
Not a css expert so if someone has a better idea how to achieve it (i don't like the `calc(100%-17rem)` TBH), you're more than welcome to add a commit here or create a new PR.

### Screenshot/screencast of this PR

fixed_next_onboarding

### What issues does this PR fix or reference?

it resolves #3656 

### How to test this PR?

1. start the onboarding and resize the window when you see the next button

### CODE REVIEW

**Comment by deboer-tim** [comment]
I'm assuming we'd also want a fixed top like the welcome screen?

I like the transparent effect, but to me it's a little too transparent right now, and I don't like how the scrollbar goes behind the bottom bar (and I don't think there's any way to really solve that). I'll defer to @mairin on all points. :)

**Comment by mairin** [comment]
Hey @lstocchi !

This is starting to look slick! A couple suggestions:

- I would only have the background of the container transparent - I would have the buttons 100% opaque.
- Maybe a little less transparent? Bump up the opacity just a little bit? (I can poke in the debugger and try to find a sweet spot and get back with it)
- It is unfortunate that the scrollbar is going under the button bar... i am wondering if the scrollbar is an object we can apply a z value to so it could be always on top?

File: packages/renderer/src/lib/onboarding/Onboarding.svelte
Comment (benoitf) [inline packages/renderer/src/lib/onboarding/Onboarding.svelte]: it's not related to this PR but IMHO we should use a color from our palette colors

@mairin what is the color that we should use instead of `#36373a`

**Comment by benoitf** [comment]
here probably should have a test ensuring that after a full scroll with a big height, we always see the buttons ?

**Comment by benoitf** [comment]
as reported by Tim, I think I would expect to see the header (Podman Setup) all the time

**Comment by benoitf** [comment]
about the `w-[calc(100%_-_17rem)]` question, why do we need to not be full width ?

https://codepen.io/benoitf/pen/xxmWEXQ

**Comment by lstocchi** [comment]
> about the `w-[calc(100%_-_17rem)]` question, why do we need to not be full width ?
> 
> https://codepen.io/benoitf/pen/xxmWEXQ

The buttons will not be visible. It's like the full width is taken in consideration of the full windows but as it does not start from the left: 0, but from left: X, then it will continue after the end of the actual window and the buttons are hidden.
I should then give a margin/padding/static position to the buttons but it seems to me a dirty solution as `w-[calc(100%_-_17rem)]`

image

**Comment by lstocchi** [comment]
> as reported by Tim, I think I would expect to see the header (Podman Setup) all the time

i'll work on it in a follow-up PR

**Comment by lstocchi** [comment]
@mairin 
fixed_next_onboarding_2

File: packages/renderer/src/lib/onboarding/Onboarding.svelte
Comment (mairin) [reply packages/renderer/src/lib/onboarding/Onboarding.svelte]: charcoal-500 for sure!

**Comment by mairin** [comment]
This looks great to me!

It might be a pain, but it might look a little better if the scrollbar bottom stops at the top border of the button bg.... but not a huge deal if not possible.

**Comment by lstocchi** [comment]
> This looks great to me!
> 
> It might be a pain, but it might look a little better if the scrollbar bottom stops at the top border of the button bg.... but not a huge deal if not possible.

🤔 i got lost with the bottom -top lol

By reading your message i got this (bottom scrollbar at the top of the button)

scroll_1

but maybe you meant this?

scroll_2

if not, can you please make me a small mockup? Thanks :)

**Comment by deboer-tim** [comment]
button bg (background), pretty sure she meant the second. :)

**Comment by mairin** [comment]
> if not, can you please make me a small mockup? Thanks :)

Whoah that was fast! Yes the second. Sorry. By buttonbg i meant the button bar bg. 🤦‍♀️ Looks great!!!

**Review by jeffmaury — CHANGES_REQUESTED** [review]
Tests are failing

**Comment by lstocchi** [comment]
@mairin 
below the result.
It's quite a css hack, because to make the content flowing behind the semi-transparent bottom bar, the content div must keep full-height but its scrollbar don't. So we need to recreate the scrollbar from scratch. I just picked charcoal-200 and charcoal-100 as colors for the scrollbar, normal state and hover. Let me know what you think
scrollbar_onboarding

**Comment by lstocchi** [comment]
> Tests are failing

they should be fixed now

**Comment by deboer-tim** [comment]
The code is getting a little 'custom' to me, and I'm also starting to wonder if we should have a WizardPage component that just does fixed titlebar/footer with this scrollbar (would also fix issue #4117), and share that with the welcome page. But I also see you've thrown down the gauntlet:

> you're more than welcome to add a commit here or create a new PR.

and I likely can't do better 😂 so will end my comment here.

**Comment by lstocchi** [comment]
> The code is getting a little 'custom' to me, and I'm also starting to wonder if we should have a WizardPage component that just does fixed titlebar/footer with this scrollbar (would also fix issue #4117), and share that with the welcome page. 

Ok, after i have a working header/footer i'll extract them in a new component so it can be reused

File: packages/renderer/src/lib/onboarding/Onboarding.svelte
Comment (mairin) [inline packages/renderer/src/lib/onboarding/Onboarding.svelte:17]: When I screenshot and eyedrop the scrollbar color values I get #5c5c5c for the lighter moving piece and #34343b for the darker track background.

The colors the rest of the app has for these is #676767 for the lighter moving piece and #424242 for the darker track background. (The closest palette colors are #676767=>charcoal-100 and #424242=>charcoal-400 but those are approximately similar and not exactly the same hex codes.)

I would change to the hex values the app is already using just so the scrollbar doesn't stand out as being different (i think it strikes me as being a bit darker or somehow a bit off? which is why i checked with the eyedropper to see what the values are.)

**Review by mairin — APPROVED** [review]
Overall this looks great to me, just a quick comment on the color values for the custom scrollbar, up to you to take or leave it.

File: packages/renderer/src/lib/onboarding/Onboarding.svelte
Comment (benoitf) [inline packages/renderer/src/lib/onboarding/Onboarding.svelte]: I guess we can't add colors like that in svelte files.

it should be part of our color palette/being referenced

**Review by benoitf — DISMISSED** [review]
we should not declare CSS colors in css/svelte files

it should reference colors from the palette

if it's because you don't know how, look at https://tailwindcss.com/docs/functions-and-directives#theme

for example to reference our purple color `theme(colors.purple.500)`

to adjust the opacity of a color 

```css
  background-color: theme(colors.blue.500 / 75%);
```

etc

File: packages/renderer/src/lib/onboarding/Onboarding.svelte
Comment (lstocchi) [reply packages/renderer/src/lib/onboarding/Onboarding.svelte:17]: When i tried using charcoal-400 it looked weird so i went to look the exact color of the scrollbar in the troubleshooting. My newbie way to take them was to screenshot the scrollbar in all its state (normal, hover, active) and using the palette. I got these colors:
normal: #676767
hover: #7b7b7b
active: a1a1a1

Following your method i picked these
normal: charcoal-100 (#707073)
hover: charcoal-50 (#767676)
active: gray-700 (#aaabac)

The result/comparison is this one. WDYT?
scrollbar_color

Comment (mairin) [reply packages/renderer/src/lib/onboarding/Onboarding.svelte:17]: Ah you are right that looks weird... I'd go back to the ones you eyedropped!

Comment (mairin) [reply packages/renderer/src/lib/onboarding/Onboarding.svelte:17]: (or I can do some picks from the palette in a mockup to test to make sure it doesn't look weird...)

**Comment by lstocchi** [comment]
I would say it's ready for a new review. Waiting for the final ok from Mo about the scrollbar colors but the rest should be done.

File: packages/renderer/src/lib/onboarding/Onboarding.svelte
Comment (jeffmaury) [inline packages/renderer/src/lib/onboarding/Onboarding.svelte:2]: Can't we have this extracted ?

File: packages/renderer/svelte.config.js
Comment (jeffmaury) [inline packages/renderer/svelte.config.js:30]: What is the rationale for this ? Any relation with this PR ?

**Comment by mairin** [comment]
Hey @lstocchi ! 

I think the gradient on the scrollbar is a perfect solution. However, it looks like it goes a little past where the background div behind the buttons on the bottom are -

Screenshot from 2023-10-05 06-53-20

It's so super faint, on one of my monitors I can't see it but on the laptop lcd (which is a little brighter) it's pretty noticeable. Is there any way to cut it off at the div it's running into?

(Also - I don´ t know how reproducible this is - I'm on linux so i can only get the first screen that says Podman is already installed, no vm to delete... so I'm squishing the window down and scaling thing up to trigger the scrollbar. Maybe not a realistic scenario, so maybe the overflow on the gradient wouldn't likely happen in practice?)

File: packages/renderer/svelte.config.js
Comment (lstocchi) [reply packages/renderer/svelte.config.js:30]: This is needed because the syntax `theme(colors...)` is not actual css so it needs to be interpreted before it runs and we need a preprocessor like postcss.

File: packages/renderer/src/lib/onboarding/Onboarding.svelte
Comment (lstocchi) [reply packages/renderer/src/lib/onboarding/Onboarding.svelte:2]: I gave a look but it seems not possible :/ 
The code related to postcss needs to be here.

My knowledge about preprocessing is not that vast, i was expecting that if i created a file with extension `.pcss` it would trigger postcss but it does not work apparently. I cannot create a simple .css file.

If someone has any suggestions i'm all ears.

**Comment by lstocchi** [comment]
> It's so super faint, on one of my monitors I can't see it but on the laptop lcd (which is a little brighter) it's pretty noticeable. Is there any way to cut it off at the div it's running into?
> 
> (Also - I don´ t know how reproducible this is - I'm on linux so i can only get the first screen that says Podman is already installed, no vm to delete... so I'm squishing the window down and scaling thing up to trigger the scrollbar. Maybe not a realistic scenario, so maybe the overflow on the gradient wouldn't likely happen in practice?)

😱 giving a look

Update: @mairin just guessing because i'm not really sure. My monitors suck and even if i put the brightness to the max i could not see it clearly lol.
My guess is that the problem is that the scrollbar has a shadow and the height of the scrollbar is the same as its parent div, and bc it goes behind the footer, also the scollbar shadow goes behind and this is what you see.

I cannot change the height of the scrollbar if i don't change the height of the parent div ... and this cannot be done otherwise the content will not scroll behind the footer. 
So, the only thing i came up is to add a fake div between the scrollbar and the footer.
The other hack could be to only add the shadow to the visible part of the scrollbar but i'm not that expert in css so i don't know how to do it.

Can you please give it a try? If we're lucky it should work 🤞

**Comment by mairin** [comment]
> Can you please give it a try? If we're lucky it should work 🤞

It worked, no more drop shadow escaping! :tada:

**Comment by lstocchi** [comment]
rebasing for the new ci workflow to run

**Review by jeffmaury — APPROVED** [review]
LGTM

**Comment by lstocchi** [comment]
@benoitf could you please give it a look? i need your approval as you asked for some changes. Thanks!!

**Review by benoitf — APPROVED** [review]
changes have been applied, no more comments

File: packages/renderer/svelte.config.js
Comment (benoitf) [inline packages/renderer/svelte.config.js]: maybe remove this trace ?

**Review by benoitf — COMMENTED** [review]
if you can remove `console.log('dirname' + __dirname);` before merging

File: packages/renderer/svelte.config.js
Comment (lstocchi) [reply packages/renderer/svelte.config.js]: done 👍

**Pull request merged**
